build-tools/packages/build-tools/src/common/fluidUtils.ts

```diff
@@ -36,67 +35,67 @@ async function isFluidRootLerna(dir: string) {
 	return true;
 }
 
-async function isFluidRootPackage(dir: string) {
+async function isFluidRootPackage(dir: string, log?: Logger) {
 	const filename = path.join(dir, "package.json");
 	if (!existsSync(filename)) {
-		verbose(`InferRoot: package.json not found`);
+		log?.verbose(`InferRoot: package.json not found`);
 		return false;
 	}
 
 	const parsed = await readJson(filename);
 	if (parsed.name === "root" && parsed.private === true) {
 		return true;
 	}
-	verbose(`InferRoot: package.json not matched`);
+	log?.verbose(`InferRoot: package.json not matched`);
 	return false;
 }
 
 async function isFluidRoot(dir: string) {
 	return (await isFluidRootLerna(dir)) && (await isFluidRootPackage(dir));
 }
 
-async function inferRoot() {
-	return lookUpDirAsync(process.cwd(), async (curr) => {
-		verbose(`InferRoot: probing ${curr}`);
-		try {
-			if (await isFluidRoot(curr)) {
-				return true;
-			}
-			// eslint-disable-next-line no-empty
-		} catch {}
-		return false;
-	});
+async function inferRoot(log?: Logger) {
+	const fluidConfig = findUp.sync("fluidBuild.config.cjs", { cwd: process.cwd(), type: "file" });
+	if (fluidConfig === undefined) {
+		return undefined;
+	}
+	const isRoot = await isFluidRootPackage(path.dirname(fluidConfig), log);
+	if (isRoot) {
+		return path.dirname(fluidConfig);
+	}
+
+	return undefined;
 }
 
-export async function getResolvedFluidRoot() {
+export async function getResolvedFluidRoot(log?: Logger) {
```

Comment:
Are we tracking updates to the callers so they start providing a logger? I see several call sites.

Reply:
I updated the signatures to use the default logger unless one is provided.